### ISSUE TITLE
Deprecated EventHandler methods on Paper starting with 1.18.1

### ISSUE THREAD

**Issue by SlimeDog** [MLG-Fortress/ExtraHardMode#303]
Paper 1.18.1-161 and ff
ExtraHardMode

On server start-up
```
[15:35:36] [Server thread/WARN]: @EventHandler method com.extrahardmode.features.AntiGrinder#onEntityDeath returns non-void type boolean. This is an unsupported behavior which will be removed in a future version of Paper.This should be reported to the developers of ExtraHardMode v3.15.0-e9c12af, (RoboMWM,Diemex,Big_Scary)
[15:35:36] [Server thread/WARN]: @EventHandler method com.extrahardmode.features.AntiGrinder#onEntitySpawn returns non-void type boolean. This is an unsupported behavior which will be removed in a future version of Paper.This should be reported to the developers of ExtraHardMode v3.15.0-e9c12af, (RoboMWM,Diemex,Big_Scary)
```

**Comment by SlimeDog** [comment]
Reminder that these are still reported by Paper 1.19.2.

**Comment by Remski01** [comment]
Have asked in the Paper discord about this error, reply:
```
an EventHandler must return void
[17:46]
or, rather, should. non-void returns may be broken in the future and are not supported.
```

**Comment by SlimeDog** [comment]
Looking at the ExtraHardMode sources, onEntityDeath() and onEntitySpawn() are declared void in all cases except in `src/main/java/com/extrahardmode/features/AntiGrinder.java`, where they are declared boolean and return true in cases that do not meet certain criteria. Are the return values used anywhere?

**Comment by SlimeDog** [comment]
I tried the fix locally, and it seems to work fine. I do not feel competent to touch the official code, so I did not make a PR. Note that the `assertFalse` and `assertTrue` tests in `src/test/com/extrahardmode/features/TestAntiGrinder.java` will have to be changed as well.

**Comment by SlimeDog** [comment]
There is a new PR that covers this issue, both the code and the tests: https://github.com/MLG-Fortress/ExtraHardMode/pull/316

**Comment by RoboMWM** [comment]
closed by #316 

ah I see why this wasn't auto-closed, it wasn't linked by the PR author.

**Comment by SlimeDog** [comment]
Confirmed fixed with the associated PR. Thanks.
